### PR TITLE
Server Connector tools for JMS, File (FTP and FS)

### PR DESCRIPTION
**Pull request by NatashaWso2**
Fixes https://github.com/ballerinalang/composer/issues/4335
Fixes https://github.com/ballerinalang/composer/issues/4336

### CODE REVIEW

File: modules/web/js/ballerina/diagram2/views/default/components/nodes/resource-node.jsx [hunk 1]
```diff
@@ -107,9 +107,19 @@ class ResourceNode extends React.Component {
         tLinkBox.y += annotationBodyHeight;
         const thisNodeIndex = parentNode.getIndexOfResources(this.props.model);
         const resourceSiblings = parentNode.getResources();
-        let showAddResourceBtn = true;
-        if (parentNode.getProtocolPackageIdentifier().value === 'ws' && resourceSiblings.length >= 6) {
-            showAddResourceBtn = false;
+        // For Web sockets
+        let showAddResourceBtnForWS = true;
+        if (parentNode.getProtocolPackageIdentifier().value === 'ws' &&
+            resourceSiblings.length >= 6) {
```
Comment (hemikak) [inline modules/web/js/ballerina/diagram2/views/default/components/nodes/resource-node.jsx]: Cant we bring this to line 112 ? To be on the earlier line .

File: modules/web/js/ballerina/tool-palette/tool-palette-view.jsx [hunk 2]
```diff
@@ -324,6 +325,68 @@ class ToolPaletteView extends React.Component {
...
+                    newToolItem = {
+                        id: 'jms-service',
+                        name: 'JMS Service',
+                        icon: 'service',
+                        title: 'JMS Service',
+                        nodeFactoryMethod: DefaultNodeFactory.createJMSServiceDef,
+                        description: 'JMS container of resources, each of which defines the logic for'
+                    + ' handling one type of request',
+                    };
+                    defaultTools.tools.forEach((elem, index) => {
```
Comment (hemikak) [inline modules/web/js/ballerina/tool-palette/tool-palette-view.jsx]: If this is to find the index, we can use findIndex method of ES6.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findIndex

File: modules/web/js/ballerina/tool-palette/tool-palette-view.jsx [hunk 3]
```diff
@@ -324,6 +325,68 @@ class ToolPaletteView extends React.Component {
...
+                    newToolItem = {
+                        id: 'fs-service',
+                        name: 'FS Service',
+                        icon: 'service',
+                        title: 'FS Service',
+                        nodeFactoryMethod: DefaultNodeFactory.createFSServiceDef,
+                        description: ' FS Server Connector can be used to listen to a ' +
+                        'directory in the local file system',
+                    };
+                    defaultTools.tools.forEach((elem, index) => {
```
Comment (hemikak) [inline modules/web/js/ballerina/tool-palette/tool-palette-view.jsx]: If this is to find the index, we can use findIndex method of ES6.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findIndex

File: modules/web/js/ballerina/tool-palette/tool-palette-view.jsx [hunk 4]
```diff
@@ -324,6 +325,68 @@ class ToolPaletteView extends React.Component {
...
+                else if (packageDefintion.getName() === 'ballerina.net.ftp') {
+                    newToolItem = {
+                        id: 'ftp-service',
+                        name: 'FTP Service',
+                        icon: 'service',
+                        title: 'FTP Service',
+                        nodeFactoryMethod: DefaultNodeFactory.createFTPServiceDef,
+                        description: 'FS Server Connector can be used to listen to a remote directory',
+                    };
+                    defaultTools.tools.forEach((elem, index) => {
```
Comment (hemikak) [inline modules/web/js/ballerina/tool-palette/tool-palette-view.jsx]: If this is to find the index, we can use findIndex method of ES6.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findIndex

File: modules/web/js/ballerina/diagram2/views/default/components/nodes/resource-node.jsx (hunk 1 above)
Comment (NatashaWso2) [reply modules/web/js/ballerina/diagram2/views/default/components/nodes/resource-node.jsx]: Fixed with 7907698b7efdcf1142c2ba9a166ea8e0ff63c5b0

File: modules/web/js/ballerina/tool-palette/tool-palette-view.jsx (hunk 2 above)
Comment (NatashaWso2) [reply modules/web/js/ballerina/tool-palette/tool-palette-view.jsx]: Fixed with 7907698

File: modules/web/js/ballerina/tool-palette/tool-palette-view.jsx (hunk 3 above)
Comment (NatashaWso2) [reply modules/web/js/ballerina/tool-palette/tool-palette-view.jsx]: Fixed with 7907698

File: modules/web/js/ballerina/tool-palette/tool-palette-view.jsx (hunk 4 above)
Comment (NatashaWso2) [reply modules/web/js/ballerina/tool-palette/tool-palette-view.jsx]: Fixed with 7907698

**Pull request merged**